corehq/apps/reports/tasks.py

```diff
@@ -169,6 +169,12 @@ def export_all_rows_task(ReportClass, report_state, recipient_list=None, subject
     report_start = datetime.utcnow()
     file = report.excel_response
     report_class = report.__class__.__module__ + '.' + report.__class__.__name__
+
+    if report.domain is None:
+        # Some HQ-wide reports (e.g. accounting/smsbillables) will not have a domain associated with them
+        # This uses the user's first domain to store the file in the blobdb
+        report.domain = report.request.couch_user.get_domains()[0]
```

Comment:
does `report.request.couch_user.get_domains()` maintains the order based on created it?

Reply:
I'm not  sure. I was just re-using the same logic. It doesn't particularly matter in this case, because no matter what the download is, it isn't related to a domain. The domain requirement is likely just a short-sighted requirement that was introduced (and kind of hard to remove)

Reply:
It may make sense to use a hard-coded/constant value here. Something like `"<hq-reports>"` (should be an illegal domain name so the blob storage used by these reports do not get attributed to a domain).

Reply:
Initially I did wonder about how to or if we should allow the blob db to have support for blobs that are not associated with a domain. If we were going to go that route, we'd also need to create a separate endpoint for those reports because these report downloads email links like `/a/<domain>/download/xxx` (I think that's true but its been a few days since I looked at it). I'm also not a huge fan of adding a separate constant for the database when `NULL` would make more sense semantically. Alternatively if we made a decision that non-domain based exports would never need to be partitioned there could be a separate un-sharded model. Though that certainly has its own issues/downsides.

I ended up coming up with this solution when I went back to figure out how this was ever working, so I decided to restore the previous behavior

Reply:
> I'm also not a huge fan of adding a separate constant for the database when `NULL` would make more sense semantically.

The main purpose of the `domain` field is for analysis reasons, to figure out which domains are using the most blob storage. For that purpose, a constant value like `<hq-reports>` would be more useful than `NULL`.

I don't think it's really used anywhere else; `parent_id` is the more important field, although even that is sometimes filled with a dummy value if there is no logical parent id.